tests/AbstractQueryBuilderTest.php

```diff
@@ -476,7 +462,16 @@ public function testBuildLikeCondition(
             . (empty($expected) ? '' : ' WHERE ' . DbHelper::replaceQuotes($expected, $db->getDriverName())),
             $sql
         );
-        $this->assertSame($expectedParams, $params);
+        $this->assertSame(array_keys($expectedParams), array_keys($params));
+        foreach ($params as $name => $value) {
+            if ($value instanceof Param) {
+                $this->assertInstanceOf(Param::class, $expectedParams[$name]);
+                $this->assertSame($expectedParams[$name]->getValue(), $value->getValue());
+                $this->assertSame($expectedParams[$name]->getType(), $value->getType());
+            } else {
+                $this->assertSame($expectedParams[$name], $value);
+            }
```

Comment:
```suggestion
            $this->assertEquals($expectedParams[$name], $value);
```
Could be replaced with `assertEquals()`

Reply:
`assertEquals` don't check types. Integer `3` equals to string `"3"`. In this case we should check parameter value type also.

Reply:
In our case it is always string, but if it is necessary - ok